projects/canopy/src/lib/forms/sort-code/sort-code.component.ts

```diff
@@ -158,18 +164,38 @@ export class LgSortCodeComponent implements OnInit, ControlValueAccessor {
   }
 
   validate(): ValidationErrors {
-    const errors: ValidationErrors = {};
+    const invalidFields: Array<string> = [];
     Object.keys(this.sortCodeFormGroup.controls).forEach((fieldName) => {
       if (
         this.sortCodeFormGroup.controls[fieldName].invalid &&
         (this.sortCodeFormGroup.controls[fieldName].dirty ||
           (this.formGroupDirective && this.formGroupDirective.submitted))
       ) {
-        errors.invalidField = true;
+        invalidFields.push(fieldName);
       }
     });
-    if (Object.keys(errors).length) {
-      return errors;
+
+    const invalidFieldNames: Array<string> = [];
+    const requiredFieldNames: Array<string> = [];
+    invalidFields.forEach((fieldName) => {
+      if (this.sortCodeFormGroup.controls[fieldName].hasError('required')) {
+        requiredFieldNames.push(labelFieldMap[fieldName]);
```

Comment:
I haven't worked on this component much so lack a bit of context, but I was a bit confused by the `labelFieldMap`, the keys and values are the same do we need the map?

Reply:
Completely right, I just took what was on the date component but I should have checked ;) I'll update it.

Reply:
Fixed 👍

Reply:
lol, I probably wrote the date component part!